Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -128,6 +128,8 @@ def main(fd):
                         cache.add(name)
                     elif cmd == b'UNREGISTER':
                         cache.remove(name)
+                    elif cmd == b'PING':
+                        os.write(fd_write, b"PONG\n")
```

Comment:
`os.write` may not write all bytes. You probably want to wrap `fd_write` in a buffered writer as is done for `fd` above.